### ISSUE TITLE
Bug: inline rule overrides via eslint-disable-next-line are ignored during linting

### ISSUE THREAD

**Issue by flq** [eslint/eslint#16230]
### Environment

Node version: 16.13.2
npm version: 8.1.2 (but actually using yarn 1.22.15)
Local ESLint version: Not found (but via react scripts version 5.0.1 version 8.18.0)
Global ESLint version: Not found
Operating System: macOS 12.15.1 (darwin 21.6.0)

Through react scripts both parsers are present:
* "@babel/eslint-parser@^7.16.3"
* "@typescript-eslint/parser@^5.5.0"

### What parser are you using?

@babel/eslint-parser

### What did you do?

Upgraded to react scripts 5. Since then inline rule overrides using e.g. "eslint-disable-next-line" are **ignored**. So, when running `eslint src` the warnings ignored through those lines are shown.

<details>
<summary>Configuration</summary>

```
{
  "extends": ["react-app", "react-app/jest"],
  "ignorePatterns": ["__mocks__"],
  "rules": {
    "react-hooks/exhaustive-deps": 0,
    "jsx-a11y/iframe-has-title": 0,
    "jsx-a11y/anchor-is-valid": 0,
    "jsx-a11y/anchor-has-content": 0,
    "jsx-a11y/alt-text": 0,
    "import/no-anonymous-default-export": 0,
    "@typescript-eslint/no-unused-vars": [
      "warn",
      { "varsIgnorePattern": "[dD]ebug|_", "argsIgnorePattern": "_", "ignoreRestSiblings": true }
    ],
    "@typescript-eslint/consistent-type-assertions": 0,
    "quote-props": ["warn", "as-needed"],

    //
    // The following rules has been turned warnings becase their enforcement
    // can improve our following of best practices
    //

    // Prefer `findByText` query over using `waitFor` + `getByText`
    "testing-library/prefer-find-by": "warn",

    // Avoid destructuring queries from `render` result, use `screen.queryByTestId` instead
    "testing-library/prefer-screen-queries": "warn",

    // Avoid wrapping Testing Library util calls in `act`
    "testing-library/no-unnecessary-act": "warn",

    // Forbidden usage of `render` within testing framework `beforeEach` setup
    "testing-library/no-render-in-setup": "warn",

    // Avoid using side effects within `waitFor` callback
    "testing-library/no-wait-for-side-effects": "warn",

    // Use `queryBy*` queries rather than `getBy*` for checking element is NOT present
    "testing-library/prefer-presence-queries": "warn",

    // Prefer using queryBy* when waiting for disappearance
    "testing-library/prefer-query-by-disappearance": "warn",

    // Unexpected debug statement
    "testing-library/no-debugging-utils": "warn",

    // Promise returned from `waitFor` must be handled
    "testing-library/await-async-utils": "warn",

    // Avoid calling `expect` conditionally`
    "jest/no-conditional-expect": "warn",

    // Async assertions must be awaited or returned
    "jest/valid-expect": "warn",

    // Illegal usage of `fail`, prefer throwing an error, or the `done.fail` callback
    "jest/no-jasmine-globals": "warn",

    //
    // The following rules has been turned off because they doesn't really help us
    //

    // `rendered` is not a recommended name for `render` returned value. Instead, you should destructure it, or name it using one of: `view`, or `utils`
    // 😒  we rename things how WE find them useful, turned it off.
    "testing-library/render-result-naming-convention": 0,

    // Avoid using container methods. Prefer using the methods from Testing Library, such as "getByRole()"
    // 😔  this is not always possible for us
    "testing-library/no-container": 0,

    // Avoid direct Node access. Prefer using the methods from Testing Library
    // 🤷‍♂️  this rule aims to disallow DOM traversal using native HTML methods and properties,
    //     but it's not really useful and sometimes we just neet to be able to do that
    "testing-library/no-node-access": 0,

    // promise returned from `findByIds` query must be handled
    // 👍  we handle those promises in a way that, for some reason, is not understood by the eslint parser
    //     check line 54 and 66 of the topics.spec.ts
    "testing-library/await-async-query": 0,

    // Using propTypes from another component is not safe because they may be removed in production builds
    // 😔  we must follow use Semnatic UI prop types for our icons, more info in AheadIcon.tsx
    "react/forbid-foreign-prop-types": 0
  },
  "overrides": [
    {
      "files": ["**/__tests__/hooks/*.spec.*"],
      "rules": {
        "react-hooks/rules-of-hooks": 0
      }
    },

    // this rule prevents the complaining around the use of "self" in the service worker
    {
      "files": ["tom.ts"],
      "rules": {
        "no-restricted-globals": 0
      }
    }
  ]
}
```
</details>

### What did you expect to happen?

eslint rules deactivated via "eslint-disable-next-line" should not show up when linting the source code.

### What actually happened?

```
/Users/.../EmbedBuildingBlockComponent.tsx
  30:13  warning  Using target="_blank" without rel="noreferrer" (which implies rel="noopener") is a security risk in older browsers: see https://mathiasbynens.github.io/rel-noopener/#recommendations  react/jsx-no-target-blank
```

EmbedBuildingBlockComponent lines 26-31:

```
{readOnly && (
    // eslint-disable-next-line react/jsx-no-target-blank
    <a className="font-size--small isHalfElementSpaced link--hover-underline is-block" href={url} target="_blank">
      {t("buildingblock.embed.openLink")}
    </a>
  )}
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

* we are using `craco` to wrap our react scripts to configure hot reloading in combination with our backend set-up.
* we tried to reproduce from a basic create-react-app with typescript setup, and there it works. In this basic setup, also both parsers are referenced.

**Comment by mdjermanovic** [comment]
Hi @flq! Thanks for the issue and sorry for the lack of response.

> ```
> /Users/.../EmbedBuildingBlockComponent.tsx
>   30:13  warning  Using target="_blank" without rel="noreferrer" (which implies rel="noopener") is a security risk in older browsers: see https://mathiasbynens.github.io/rel-noopener/#recommendations  react/jsx-no-target-blank
> ```
> 
> 
>     
> 
>   
> 
> EmbedBuildingBlockComponent lines 26-31:
> 
> 
> ```
> {readOnly && (
>     // eslint-disable-next-line react/jsx-no-target-blank
>     <a className="font-size--small isHalfElementSpaced link--hover-underline is-block" href={url} target="_blank">
>       {t("buildingblock.embed.openLink")}
>     </a>
>   )}
> ```

By this info, it seems that you're getting the `react/jsx-no-target-blank` error on line 30 (`</a>`), while the disable comment applies to line 28 (`<a className=...`). Could you provide a repro repo? Otherwise, it would be very difficult to figure out what is exactly causing this problem (it could be the rule, parser, or some preprocessing).

**Comment by flq** [comment]
Yeah...we did actually try to reproduce but this isn't happening on a small repro. I know this is wholly unsatisfactory (imagine how it feels for us, considering you don't find a problem like that on google), but we were hoping for any pointers into which direction we could investigate.

E.g. (fully speculative):
* the code linting works (we get warnings and all)
* config works
* disabling based on code comments doesn't

So it could be that for whatever reason the linter only gets to see a "version" where all comments are stripped. What could possibly cause this?

**Comment by mdjermanovic** [comment]
> So it could be that for whatever reason the linter only gets to see a "version" where all comments are stripped.

It could be that the comments are stripped, but it looks more like the lines are different so the error is reported on a line that is not next to the `// eslint-disable-next-line react/jsx-no-target-blank` comment. Could you run ESLint with [json formatter](https://eslint.org/docs/latest/user-guide/formatters/#json) (`-f json`) and find the `"source"` key in the output for that file? The value should be the source code that was passed to the linter.

**Comment by flq** [comment]
Okay, when I do that for the component in question, I get the following

> I copy/pasted the source-property, and replaced \n with new line and \" with "

```tsx
import React from "react";
import { AheadIcon } from "../../../../components";
import { BuildingBlockComponentProps } from "../../building-block";
import { EmbedBuildingBlockContent } from "./EmbedBuildingBlockContent";
import { BuildingBlockTitleComponent } from "../components";
import { useTranslation } from "react-i18next";

export type EmbedBuildingBlockComponentProps = BuildingBlockComponentProps<EmbedBuildingBlockContent>;

export function EmbedBuildingBlockComponent({
  nodeKey,
  readOnly,
  content: { title, url, heightPercent = 55 },
  onContentChange,
}: EmbedBuildingBlockComponentProps) {
  const { t } = useTranslation();
  return (
    <>
      <BuildingBlockTitleComponent id={`${nodeKey}-title`} title={title} readOnly={readOnly} onContentChange={onContentChange} />
      {url ? (
        <>
          <figure className="iframeContainer" style={{ paddingBottom: `${heightPercent}%` }}>
            <iframe className="iframeContainer__iframe" src={url}></iframe>
          </figure>

          {readOnly && (
            <a className="font-size--small isHalfElementSpaced link--hover-underline is-block" href={url} target="_blank">
              {t("buildingblock.embed.openLink")}
            </a>
          )}
        </>
      ) : (
        <div className="video__placeholder">
          <AheadIcon name="page" className="placeholder__icon" />
        </div>
      )}
    </>
  );
}
```

So if you expected the comment to be there, it isn't.

**Comment by mdjermanovic** [comment]
Can you check if you are running ESLint with [`--no-inline-config`](https://eslint.org/docs/latest/user-guide/command-line-interface#--no-inline-config) flag? Or, in the case that your setup runs ESLint through [API](https://eslint.org/docs/latest/developer-guide/nodejs-api#-new-eslintoptions), check if you have `allowInlineConfig: false` somewhere in your configurations (perhaps in the `craco` config under `eslint.pluginOptions`). You could also check if you have `noInlineConfig: true`  (though, this would log a warning on the use of `// eslint-disable-next-line` comment).

**Comment by snitin315** [comment]
No response, hence closing.
